scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,302 @@
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local nth_input_file="$1"
+    IFS=" " read -r -a all_input_files <<< "$2"
+
+    if $SOLC --bin "$nth_input_file" "${all_input_files[@]}" > /dev/null 2>&1
+    then
+        ! [[ -e stderr.txt ]] || { echo "stderr.txt already exists. Refusing to overwrite."; exit 1; }
+
+        if [ "${IMPORT_TEST_TYPE}" == "ast" ]
+        then
+            # save exported json as expected result (silently)
+            $SOLC --combined-json ast --pretty-json "$nth_input_file" "${all_input_files[@]}" > expected.json 2> /dev/null
+            # import it, and export it again as obtained result (silently)
+            if ! $SOLC --import-ast --combined-json ast --pretty-json expected.json > obtained.json 2> stderr.txt
+            then
+                # For investigating, use exit 1 here so the script stops at the
+                # first failing test
+                # exit 1
+                FAILED=$((FAILED + 1))
+                echo -e "ERROR: AST reimport failed for input file $nth_input_file"
+                echo
+                echo "Compiler stderr:"
+                cat ./stderr.txt
+                echo
+                echo "Compiler stdout:"
+                cat ./obtained.json
+                return 1
+            fi
+            set +e
+            DIFF="$(diff expected.json obtained.json)"
+            set +e
+            if [ "$DIFF" != "" ]
+            then
+                if [ "$DIFFVIEW" == "" ]
+                then
+                    echo -e "ERROR: JSONS differ for $1: \n $DIFF \n"
+                    echo "Expected:"
+                    cat ./expected.json
+                    echo "Obtained:"
+                    cat ./obtained.json
+                else
+                    # Use user supplied diff view binary
+                    $DIFFVIEW expected.json obtained.json
+                fi
+                FAILED=$((FAILED + 1))
+                return 2
+            fi
+            TESTED=$((TESTED + 1))
+            rm expected.json obtained.json
+            rm -f stderr.txt
+        elif [ "${IMPORT_TEST_TYPE}" == "evm-assembly" ]
+        then
+            local types=( "asm" "bin" "bin-runtime" "opcodes" "srcmap" "srcmap-runtime" )
+            local _TESTED=1
+            if ! $SOLC --combined-json bin,bin-runtime,opcodes,asm,srcmap,srcmap-runtime --pretty-json "$nth_input_file" "${all_input_files[@]}" > expected.json 2> expected.error
+            then
+                printf "\n"
+                echo "$nth_input_file"
+                cat expected.error
+                UNCOMPILABLE=$((UNCOMPILABLE + 1))
+                return 0
+            else
+                for contract in $(jq '.contracts | keys | .[]' expected.json 2> /dev/null)
```

Comment:
Why silence stderr here? I think `jq` does not print anything there by default anyway and If something goes wrong here, we won't see the error.

Reply:
Sorry, for the late answer I needed to check again why I actually did this. The reason is simple, we have some files that only consists of free functions. So the resulting combined json files does not have any keys. To not get spammed of those errors eg. `jq: error: null (null) has no keys`, I redirected the errors to `/dev/null/`.

Reply:
Makes sense. Could you put that in a comment though? I think we should leave stderr enabled and only disable it when it's too noisy. And it helps if it's explained why it's disabled in a particular case.